src/main/java/com/melon/portfoliomanager/controllers/UserController.java

```diff
@@ -0,0 +1,59 @@
+package com.melon.portfoliomanager.controllers;
+
+
+import com.melon.portfoliomanager.dtos.UserDto;
+import com.melon.portfoliomanager.exceptions.UsernameAlreadyUsedException;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.services.UserService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+
+    @PostMapping("/add")
+    public ResponseEntity<String> createUser(@Valid @RequestBody UserDto userDto) throws Exception {
+
+        try {
+            User dtoToUser = userService.transformDto(userDto);
+            userService.createUser(dtoToUser);
+
+            return new ResponseEntity<>(HttpStatus.CREATED);
+        }
+        catch (UsernameAlreadyUsedException e){
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+        catch (Exception e) {
+            return new ResponseEntity<>("Internal Server Error.", HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
It is best to start using logger (from ```slf4``` for example) in order to be able to debug if an unexpected error occurred.